btrfs.h

```diff
@@ -15,22 +15,28 @@
 */
 
 #include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,7,0)
+#error "Headers from kernel >= 4.7 are required on compilation time (not on run time)"
+#endif
 #include <linux/magic.h>
-#include <btrfs/ioctl.h>
-#include <btrfs/ctree.h>
+#include <linux/btrfs.h>
+#include <linux/btrfs_tree.h>
 
 // unfortunately, we need to define "alignment safe" C structs to populate for
 // packed structs that aren't handled by cgo. Fields will be added here, as
 // needed.
 
 struct gosafe_btrfs_root_item {
-	u8 uuid[BTRFS_UUID_SIZE];
-	u8 parent_uuid[BTRFS_UUID_SIZE];
-	u8 received_uuid[BTRFS_UUID_SIZE];
+	uint8_t uuid[BTRFS_UUID_SIZE];
+	uint8_t parent_uuid[BTRFS_UUID_SIZE];
+	uint8_t received_uuid[BTRFS_UUID_SIZE];
 
-	u64 gen;
-	u64 ogen;
-	u64 flags;
+	uint64_t gen;
+	uint64_t ogen;
+	uint64_t flags;
```

Comment:
Somewhat immaterial to this PR, but I personally would prefer to use the kernel types here (e.g. `__le64`), even though they are just used for kernel type (read: endianness) checking with sparse as they increase readability when switching from kernel to Cgo code, and they encode the intended endianness of the fields in a way that is obvious to the programmer.

I'd also prefer to switch to the kernel names here instead of inventing unique names, for the same reason.

Reply:
done